src/main/com/yetanalytics/datasim/input/alignments.clj

```diff
@@ -1,12 +1,35 @@
 (ns com.yetanalytics.datasim.input.alignments
   (:require [clojure.spec.alpha :as s]
+            [clojure.walk :as w]
             [com.yetanalytics.datasim.protocols :as p]
             [com.yetanalytics.pan.objects.profile :as profile]
             [com.yetanalytics.datasim.iri :as iri]
             [com.yetanalytics.datasim.xapi :as xapi]))
 
+;; TODO: Object overrides
+;; Due to limitations of keywords, we cannot have IRI keys, so no extensions
+
 ;;Alignment: Map of Component and Weight
 
+(defn- no-iri-keys?
+  "Returns false if there exists a key made from an IRI, e.g.
+   (name :https://foo.org) => \"/foo.org\""
+  [obj]
+  (cond
+    (map? obj)
+    (if-not (->> obj vals (map no-iri-keys?) (some false?))
+      (->> obj keys (some (partial re-matches #".*/.*")) not)
+      false)
+    (vector? obj)
+    (reduce (fn [acc x] (if (no-iri-keys? x) acc false)) true obj)
+    :else
+    true))
```

Comment:
Even if the reason you had to write it was an annoying intersection of domain and data representation, this is a pretty neat function. As we discussed previously there are some cases in here where `clojure.core/every?` would probably simplify and make it more readable.

Reply:
Huh I must've forgotten `every?` existed. Thanks for pointing that out.